packages/astro/src/assets/vite-plugin-assets.ts

```diff
@@ -223,22 +226,31 @@ export default function assets({ settings }: { settings: AstroSettings }): vite.
 					if (settings.config.experimental.svg && /\.svg$/.test(id)) {
 						const { contents, ...metadata } = imageMetadata;
 						// We know that the contents are present, as we only emit this property for SVG files
-						return makeSvgComponent(metadata, contents!, {
-							mode: settings.config.experimental.svg.mode,
-						});
+						return {
+							code: makeSvgComponent(metadata, contents!, {
+								mode: settings.config.experimental.svg.mode,
+							}),
+							moduleType: 'js',
+						};
 					}
 
 					// We can only reliably determine if an image is used on the server, as we need to track its usage throughout the entire build.
 					// Since you cannot use image optimization on the client anyway, it's safe to assume that if the user imported
 					// an image on the client, it should be present in the final build.
 					if (options?.ssr) {
-						return `export default ${getProxyCode(
-							imageMetadata,
-							settings.buildOutput === 'server',
-						)}`;
+						return {
```

Comment:
Makes it an object that returns `code` (rollup/rolldown compatible shape) and `moduleType` (only rolldown I think)